frontend/__tests__/components/environment.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+
+import { EnvironmentPage } from '../../public/components/environment';
+
+describe('EnvironmentPage', () => {
+
+  const objects={
+    kind: '',
+    apiVersion: '',
+    metadata: {},
+    spec: {},
+    status: {}
+  };
+
+  const environmentPageRO=<EnvironmentPage
+    obj={objects}
+    rawEnvData={[ { 'env': [ { 'name': 'test', 'value': ':0', 'ID': 0 } ] } ]}
+    envPath={[]}
+    readOnly={true}
+  />;
+
+  const environmentPage=<EnvironmentPage
+    obj={objects}
+    rawEnvData={[ { 'env': [ { 'name': 'test', 'value': ':0', 'ID': 0 } ] } ]}
+    envPath={['test']}
+    readOnly={false}
+  />;
+
+  it('does not show help text when page read-only', () => {
+    const wrapper=shallow(environmentPageRO);
+    expect(wrapper.find('p').exists()).toEqual(false);
+  });
+
+  it('renders save and reload buttons when page not read-only', () => {
+    const wrapper=shallow(environmentPage);
+    expect(wrapper.find('.environment-buttons button').exists()).toEqual(true);
+  });
+
+  it('renders error message when error in state', () => {
```

Comment:
If there is not an error message in the state, do we have to prove that it will not render the error message?  Postive/negative test cases?  This comment could apply to any of these, ensuring a future update 'oops' doesn't trigger an incorrect view.

Reply:
I added positive/negative test cases for this case and the ones below in the newest commit.